src/server.c

```diff
@@ -6329,13 +6316,15 @@ sds genValkeyInfoString(dict *section_dict, int all_sections, int everything) {
         for (j = 0; j < server.dbnum; j++) {
             serverDb *db = server.db[j];
             if (db == NULL) continue;
-            long long keys, vkeys;
+            long long keys, vkeys, keysvitems;
 
             keys = kvstoreSize(db->keys);
             vkeys = kvstoreSize(db->expires);
+            keysvitems = kvstoreSize(db->keys_with_volatile_items);
+
             if (keys || vkeys) {
-                info = sdscatprintf(info, "db%d:keys=%lld,expires=%lld,avg_ttl=%lld\r\n", j, keys, vkeys,
-                                    db->avg_ttl);
+                info = sdscatprintf(info, "db%d:keys=%lld,expires=%lld,avg_ttl=%lld,keys_with_volatile_items=%lld\r\n", j, keys, vkeys,
```

Comment:
This is a little risky.  This is a highly referenced info line, and a change like this will almost certainly break compatibility with some clients which parse this data.

The name `keys_with_volatile_items` should be reviewed.  Is this intended just for hashes?  Or do we expect this to be used for other types?  If just hashes, maybe "keys" should be "hashes".  Also "items"?  Our documentation almost universally refers to these as "fields" in a hash.

Reply:
We kinda copy/borrow this from redis as they use the same for the same use case so it provide some confidence it will not break current clients. If there are some control plane logic exists which will break? Maybe? I think we can still think of introducing a separate statistic here.

>Or do we expect this to be used for other types?

Yes. I think the plan is that this will encapsulate any future data type (eg sets) here as well.